Framework/Algorithms/src/CompareWorkspaces.cpp

```diff
@@ -1143,23 +1143,33 @@ void CompareWorkspaces::doTableComparison(
   }
 
   const bool checkAllData = getProperty("CheckAllData");
-
-  for (size_t i = 0; i < numRows; ++i) {
-    const TableRow r1 =
-        boost::const_pointer_cast<ITableWorkspace>(tws1)->getRow(i);
-    const TableRow r2 =
-        boost::const_pointer_cast<ITableWorkspace>(tws2)->getRow(i);
-    // Easiest, if not the fastest, way to compare is via strings
-    std::stringstream r1s, r2s;
-    r1s << r1;
-    r2s << r2;
-    if (r1s.str() != r2s.str()) {
-      g_log.debug() << "Table data mismatch at row " << i << " (" << r1s.str()
-                    << " vs " << r2s.str() << ")\n";
-      recordMismatch("Table data mismatch");
-      if (!checkAllData)
-        return;
-    }
+  const bool relErr = getProperty("ToleranceRelErr");
+  const double tolerance = getProperty("Tolerance");
+  
+  for (size_t i = 0; i < numCols; ++i) {
+    const auto c1 =
+        boost::const_pointer_cast<ITableWorkspace>(tws1)->getColumn(i);
+    const auto c2 =
+        boost::const_pointer_cast<ITableWorkspace>(tws2)->getColumn(i);
+    
+        if(relErr){
+          if (!c1->equalsRelErr(c2.get(),tolerance)) {
```

Comment:
Any way we can reduce duplication here?  Maybe capture the return from `equals` or `equalsRelErr` and then do the common `g_log.debug()...` stuff?